lib/grpc/GrpcService.ts

```diff
@@ -431,6 +432,32 @@ class GrpcService {
     }
   }
 
+  /**
+   * See [[Service.listTrades]]
+   */
+  public listTrades: grpc.handleUnaryCall<xudrpc.ListTradesRequest, xudrpc.ListTradesResponse> = async (call, callback) => {
+    try {
+      const listTrades = await this.service.listTrades(call.request.toObject());
+      const response = new xudrpc.ListTradesResponse();
+      const trades: xudrpc.Trade[] = [];
+      listTrades.forEach(async (trade: TradeInstance) => {
+        const grpcTrade = new xudrpc.Trade();
+        const order = grpcTrade.getMakerOrder();
+        const deal = await trade.getSwapDeal();
```

Comment:
It looks like they only thing we're using the `deal` for is the pair id, which we're getting anyway from the orders. We can probably drop this line and if you make my changes above instead use `grpcTrade.setPairId(makerOrder!.pairId);` below.